snapshot-labs/snapshot
# chore: add test to CI

### Issues

Tests are not currently being ran on github actions

### Changes 

1. Add a new tests task to github actions, running the already existing tests
2. Upload the test coverage to the organization codecov account
3. Change the test config, to base the test coverage on all files, and not only tested files 
4. Upgrade test related dependencies (one notable change is the move from the deprecated c8 to v8 for code coverage)
5. Remove unused tests package (vitest-fetch-mock), which was not working with latest vitest version

### How to test

1. Run `yarn test:unit:coverage`: it should generate a coverage report based on all files, and not only tested files. Coverage drop from ~78% to ~5%, which is more representative of the state of the tests
2. Push a commit: the `test` task on github actions should run

See the github actions task for this PR for example


### Self-review checklist

### Additional notes or considerations

- No additional tests have been added/removed
- E2E tests are failing
